### DISCUSSION TITLE
How to remove all subscriptions without profiles from All Channels, leaving only subscriptions with profiles?

### DISCUSSION THREAD

**Discussion by aleksejrs** [FreeTubeApp/FreeTube#3777] [Q&A]
I decided to separate some of my profiles into a separate instance of FreeTube. I deleted the profiles I didn't need in the new instance, but the subscriptions from them are still in All Channels.

Something while doing that also caused the All Channels line in `profiles.db` to be duplicated. If I remove all the copies, FreeTube breaks; if I empty the only copy, All Channels is empty and not repopulated with the subscriptions in profiles.

**Comment by aleksejrs** [comment]
A bad, inefficient, manual solution (Backup first! For one, replacing `},{` with `},\r{` in `playlists.db` makes FT ignore its contents):
First, consider whether you need more than 50% of the subscriptions that are in All Channels in one of the instances, or less.

1. If it is more than 50%, have the full `allChannels` line in `profiles.db`. If you've deleted the profiles already, you can copy the relevant lines over from the old profile. Note that the file might contain backups or something, especially of `allChannels`.
2. Make All Channels the default profile.
3. Unsubscribe from the most obvious irrelevant channels in All Channels, unless you have so many that step 4 is faster.
4. Go to the Channels tab of one of the profiles with channels you don't need in this instance, middle-click a channel near the right end of the top row (or where the Unsubscribe button will appear). It will be opened in a new window in the All Channels profile. Click unsubscribe. If it is not subscribed, click Subscribe and the Unsubscribe to unsubscribe in the original profile.

.

1. If it is less than 50%, empty the list of subscriptions in the allChannels line (do not delete the whole line, but only the part inside the `[]`).
2. Make All Channels the default profile.
3. Go to the Channels tab of one of the profiles with channels you need in this instance, middle-click each channel and subscribe. Do not open more than one row of channels, or FT will freeze at 30 or 40 windows.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> I decided to separate some of my profiles into a separate instance of FreeTube. 

Just to clarify for my brain, with separate instance u mean a separate install?

> I deleted the profiles I didn't need in the new instance, but the subscriptions from them are still in All Channels.

This is expected as the All Channels profile reflects all channels u subscribed to across all profiles
 
> Something while doing that also caused the All Channels line in `profiles.db` to be duplicated. 

So this happened when deleting profiles?

**Reply by aleksejrs** [reply]
> Just to clarify for my brain, with separate instance u mean a separate install?

I mean running FreeTube with a different Debian system account.

> This is expected as the All Channels profile reflects all channels u subscribed to across all profile

[This](https://docs.freetubeapp.io/usage/profiles/) is a better description, because that's what *happens* and it does not look at expectations:
> If that channel isn’t in your “All Channels” profile, then FreeTube will automatically add that channel there as well.

Since I am not subscribed to some channel in any other profile, the All Channels profile does not reflect the state of the other profiles for that channel. So it is "you *have* subscribed to" and not "you *are* subscribed to".


>>    Something while doing that also caused the All Channels line in profiles.db to be duplicated.
>
> So this happened when deleting profiles?


I may have edited the file manually before that happened, but I might not have edited my main profile, and it contains multiple similar instances of allChannels of different length. Maybe those are backups?

**Comment by absidue** [comment]
> I deleted the profiles I didn't need in the new instance, but the subscriptions from them are still in All Channels.

That's intentional, the all channels profile can have it's own subscriptions too, lots of users don't create separate profiles.

If you unsubscribe from a channel inside a profile it only removes it from that profile, if you unsubscribe from a channel in the all channels profile, it gets removed from all profiles.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@aleksejrs do you still have some questions that are feel unanswered to u?

**Comment by aleksejrs** [comment]
Well, that means https://github.com/FreeTubeApp/FreeTube/discussions/3777#discussioncomment-6461845 is the only solution.
